source/pool/contracts/Pool.sol

```diff
@@ -116,34 +174,105 @@ contract Pool is Ownable {
    * @param token IERC20
    */
   function withdraw(Claim[] memory claims, IERC20 token) external {
-    withdrawProtected(claims, token, 0);
+    withdrawProtected(claims, token, 0, msg.sender, msg.sender);
+  }
+
+  /**
+   * @notice Withdraw tokens from the pool using claims and send to recipient
+   * @param claims Claim[]
+   * @param token IERC20
+   * @param recipient address
+   */
+  function withdrawWithRecipient(
+    Claim[] memory claims,
+    IERC20 token,
+    address recipient
+  ) external {
+    withdrawProtected(claims, token, 0, msg.sender, recipient);
+  }
+
+  /**
+   * @notice Withdraw tokens from the pool using claims and stake
+   * @param claims Claim[]
+   * @param token IERC20
+   */
+  function withdrawAndStake(Claim[] memory claims, IERC20 token) external {
+    require(token == IERC20(stakingToken), "INVALID_TOKEN");
+    uint256 amount = withdrawProtected(
+      claims,
+      token,
+      0,
+      msg.sender,
+      msg.sender
+    );
+    (bool success, ) = address(stakingContract).call(
+      abi.encodeWithSignature("stakeFor(address,uint256)", msg.sender, amount)
+    );
+    require(success, "ERROR_STAKING");
+  }
+
+  /**
+   * @notice Withdraw tokens from the pool using claims and stake for another account
+   * @param claims Claim[]
+   * @param token IERC20
+   * @param account address
+   */
+  function withdrawAndStakeFor(
+    Claim[] memory claims,
+    IERC20 token,
+    address account
+  ) external {
+    require(token == IERC20(stakingToken), "INVALID_TOKEN");
+    uint256 amount = withdrawProtected(
+      claims,
+      token,
+      0,
```

Comment:
Should these be zero or be able to bet set by the client? Maybe `0` as a default?

Reply:
For the original withdraw function, it was coded 0 by default, so I carried that over to withdrawWithRecipient, withdrawAndStake, and withdrawAndStakeFor.
I can see that it doesn't allow for a client to set a minimum, so update that as you mention. 
I am thinking to allow this for all the withdraw functions, and then set withdrawProtected to internal, as I don't see why it would need to be called as the other withdraw functions cover the same functionality.

Reply:
The original `withdraw` was for backwards compatibility so had just set minimum to `0`. Normal usage will still be to call withdraw directly so do need a public version of it, even if shared logic is moved to an internal function.

Reply:
- Updated the new withdraw functions (withdrawWithRecipient, withdrawAndStake, withdrawAndStakeFor) with a minimumAmount parameter
- Kept the original withdraw function as is for backwards compatibility and maintained the withdrawProtected function as public